File: core/src/main/kotlin/com/github/haschi/dominium/haushaltsbuch/core/model/values/Inventar.kt
package com.github.haschi.dominium.haushaltsbuch.core.model.values

import java.time.LocalDateTime

data class Inventar(
        val erstelltAm: LocalDateTime,
        val anlagevermoegen: Vermoegenswerte,
        val umlaufvermoegen: Vermoegenswerte,
        val schulden: Vermoegenswerte)
{

    val reinvermoegen: Reinvermögen
        get() = Reinvermögen(
                Währungsbetrag(
                        anlagevermoegen.summe.wert.add(
                                umlaufvermoegen.summe.wert)),
                schulden.summe)

    companion object
    {
        val leer: Inventar
            get() = Inventar(
                    LocalDateTime.MIN,
                    Vermoegenswerte.keine,
                    Vermoegenswerte.keine,
                    Vermoegenswerte.keine)
    }
}

File: ui/frontend/src/main/src/app/inventur/shared/inventar.ts
import { Vermoegenswert } from '../bilanz/bilanz.model';
import { Inventarposition } from './inventarposition';

export interface Reinvermoegen {
    summeDesVermoegens: string;
    summeDerSchulden: string;
    reinvermoegen: string
}

export interface Inventar {
    erstelltAm: string,
    anlagevermoegen: Inventarposition[];
    umlaufvermoegen: Inventarposition[];
    schulden: Inventarposition[];
    reinvermoegen: Reinvermoegen;
}

export interface InventurEingabe
{
    anlagevermoegen: Vermoegenswert[],
    umlaufvermoegen: Vermoegenswert[],
    schulden: Vermoegenswert[]
}